File: ruby solutions/pascal_triangle_II.rb
# @param {Integer} row_index
# @return {Integer[]}
def get_row(row_index)
    pascal = [[1],[1,1]]
    (2..row_index).each {
        |i| 
        row = [1]
        (0...i-1).each {
            |j|
            row << (pascal[i-1][j]+pascal[i-1][j+1])
        }
        row << 1
        pascal << row
    }
    return pascal[row_index]
end

File: python solutions/pascal_triangle_II.py
from typing import List


class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        pascal = [[1],[1,1]]
        if rowIndex > 1:
            for i in range(3, rowIndex+2):
                row = [1]
                for j in range(i-2):
                    row.append(pascal[i-2][j] + pascal[i-2][j+1])
                row.append(1)
                pascal.append(row)
                
        return pascal[rowIndex]